### ISSUE TITLE
Long fitting and prediction time for Linear Regressor

### ISSUE THREAD

**Issue by TamerAbdelmigid** [unit8co/darts#701]
Thanks for the impressive effort and an amazing library.

I have two issues, First, when using the LinearRegressionModel to predict a univariate problem, it takes 2.28 s for fitting compared with 2ms using sklearn directly, is this normal or am I doing something wrong?

second, to memic the behavior of sklearn predict where I feed the model the entire validation set and it produce an output for each row -regressor.predict(X_valid) - I used backtest on a series which combines train and validation sets, and set the start to 0.5 and retrain = False, last_point_only = True, This took 32 s, compared to 1ms in sklearn implementation, is there a more elegant way in implementing this?

**Comment by dennisbader** [comment]
Hi @TamerAbdelmigid and thanks for writing.

This is true, currently our RegressionModels have their bottle neck in loading the data.
This is because we use our PyTorch Dataloaders from which we iteratively extract lagged target and covariate values.

We are actually working on a performance improvement for RegressionModels which should significantly reduce computational time.

**Comment by TamerAbdelmigid** [comment]
Thanks for the clarification.
